From 7d08c98c5f6b371e0123078827353547bcbbb3df Mon Sep 17 00:00:00 2001
From: Robert Chiras <robert.chiras@nxp.com>
Date: Mon, 21 May 2018 15:12:04 +0300
Subject: [PATCH 3830/5242] MLK-18357-3: arm64: dts: Update 8M, 8QM and 8QXP
 dts files for dsi suspend/resume

commit  2bb26e01ca393c077caa50d6dfe3a701d92e9a62 from
https://source.codeaurora.org/external/imx/linux-imx.git

Added no_clk_reset property for 8M dts files, since DSI doesn't need
it's clocks stopped during suspend.
Also, added power on delay for 8QM and 8QXP for a better suspend/resume
stability.

Signed-off-by: Robert Chiras <robert.chiras@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi     |    2 ++
 arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dts  |    2 ++
 arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts |    2 ++
 3 files changed, 6 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
index c4e4164..d001ef8 100755
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
@@ -642,6 +642,7 @@
 		assigned-clock-rates = <80000000>;
 		phys = <&mipi_dsi_phy>;
 		phy-names = "dphy";
+		no_clk_reset;
 		status = "disabled";
 
 		port@0 {
@@ -666,6 +667,7 @@
 		mux-sel = <&gpr>;
 		phys = <&mipi_dsi_phy>;
 		phy-names = "dphy";
+		no_clk_reset;
 		status = "disabled";
 
 		port@0 {
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dts
index 091c4fc..56907e6 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dts
@@ -1251,6 +1251,7 @@
 };
 
 &mipi_dsi1 {
+	pwr-delay = <10>;
 	status = "okay";
 };
 
@@ -1292,6 +1293,7 @@
 };
 
 &mipi_dsi2 {
+	pwr-delay = <10>;
 	status = "okay";
 };
 
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts
index 4088434..8a5bb57 100755
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts
@@ -981,6 +981,7 @@
 };
 
 &mipi_dsi1 {
+	pwr-delay = <10>;
 	status = "okay";
 };
 
@@ -1059,6 +1060,7 @@
 };
 
 &mipi_dsi2 {
+	pwr-delay = <10>;
 	status = "okay";
 };
 
-- 
1.7.9.5

